### PR TITLE
Fix font size consistency initial

### PR DESCRIPTION
**Pull request by bratter**
First step to address #527. Submitting PR against master so you can see changes, but as per @Golodhros may want to redo this against an RC branch.

## Description
* Tweaked line chart demo to show a yAxisLabel
* Added getFontSize method to textHelpers
* For line, stacked-area, mini-tooltip, and legend
  - Added font-size scss declarations
  - Removed font sizing from chart files
  - Tweaked sizing methodologies where appropriate to dynamically adjust for font size

## Motivation and Context
#527.

## How Has This Been Tested?
* Added new test for font size getter
* All tests passed
* Visually inspected all charts confirming no unanticipated styling changes

## Types of changes
- [x] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/helpers/text.js [hunk 1]
```diff
@@ -149,11 +149,21 @@ define(function(require) {
...
         return b.measureText(text).width;
+    };
+
+    /**
+     * Gets the font size of the passed node using getComputedStyle
+     * @param {HTMLElement} node The node to get the computed font size for
+     * @return {number}
+     */
+    function getFontSize(node) {
+        return parseFloat(window.getComputedStyle(node).fontSize);
```
Comment (Golodhros) [inline src/charts/helpers/text.js]: Let's include here a check for the existence of 'window.getComputedStyle' (in Britecharts React we use Jest, that uses jsDOM and maybe doesn't have this method)

File: src/charts/line.js [hunk 2]
```diff
@@ -306,7 +306,7 @@ define(function(require){
          */
         function adjustYTickLabels(selection) {
             selection.selectAll('.tick text')
-                .attr('transform', 'translate(0, -7)');
+                .attr('dy', '-0.32em');
```
Comment (Golodhros) [inline src/charts/line.js]: Maybe we should use a const for this value?

File: src/charts/stacked-area.js [hunk 3]
```diff
@@ -579,7 +574,7 @@ define(function(require){
          */
         function adjustYTickLabels(selection) {
             selection.selectAll('.tick text')
-                .attr('transform', `translate(${yTickTextXOffset}, ${yTickTextYOffset})`);
+              .attr('dy', '-0.32em');
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: same here

File: src/styles/common/_legend.scss [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
 
     .legend-entry {
+        font-size: $text-size-medium;
+
+        .legend-entry-name,
+        .legend-entry-value,
+        .legend-circle {
+            // Ensure font-size can't be set on child elements
+            // Allowing font-size on parent to be used in calculations
+            font-size: 1em !important;
```
Comment (Golodhros) [inline src/styles/common/_legend.scss]: Let's use local variables for these two values, like in those 'enter-opacity' and 'exit-opacity' examples at the top of the file

File: src/styles/common/_legend.scss [hunk 5]
```diff
@@ -1,13 +1,29 @@
...
+            letter-spacing: 0.5px;
+        }
+
+        .legend-entry-value {
+            text-anchor: end;
+            letter-spacing: 0.8px;
```
Comment (Golodhros) [inline src/styles/common/_legend.scss]: Same here, thanks!

File: src/styles/common/_tooltip.scss [hunk 6]
```diff
@@ -15,10 +22,22 @@ $tooltip-width: 1;
...
 
+// Mini-Tooltip
+.britechart-mini-tooltip {
+    .mini-tooltip-title,
+    .mini-tooltip-name {
+        font-size: $text-size-large;
+    }
+
+    .mini-tooltip-value {
+        font-size: 27px;
```
Comment (Golodhros) [inline src/styles/common/_tooltip.scss]: Let's use a variable here.

File: src/charts/helpers/text.js (hunk 1 above)
Comment (bratter) [reply src/charts/helpers/text.js]: Just FYI, there also appears to be an unchecked getComputedStyle call in src/charts/helpers/style.js, line 31

**Comment by Golodhros** [comment]
Thanks a lot @bratter!!
This is good to go!

I won't merge it into master though. I will try to figure out how to manage an RC branch with github, as this is something we haven't done yet. Then I will merge it into that RC branch.

Thanks a lot for this work, it is an improvement in consistency that was really needed!

**Comment by bratter** [comment]
Cool. When I get a chance over the next few weeks I may continue on pulling out duplicate code and extending the changes to the other chart types.

**Comment by dalerasrorov-eb** [comment]
Great work! Thanks

**Comment by miglesiasEB** [comment]
Merged into release/3.0.0-rc1

Thanks a lot!

**Pull request closed without merging**